pyfr/integrators/base.py

```diff
@@ -181,3 +185,33 @@ def _add(self, *args):
 
         # Bind and run the axnpby kernels
         self._queue.enqueue_and_run(axnpby, *args[::2])
+
+    def grad_pvars(self, pnames, ele_index, eset):
```

Comment:
I would just have it return everything (no pnames/ele_index, or eset arg).

Reply:
Done,  I had move the function to `BaseIntegrator` to ensure compatibility with other than `std` integrators